source/trigger/PortalTeleport.cpp

```diff
@@ -21,11 +21,25 @@ void PortalTeleport::setAction(radix::Entity &trigger, const std::string &destin
   dynamic_cast<radix::entities::Trigger&>(trigger).setActionOnMove([&trigger, destination]
                                                              (radix::entities::Trigger &trigger) {
     radix::entities::Player &player = trigger.world.getPlayer();
-    if (trigger.world.destinations.find(destination)
-        != trigger.world.destinations.end()) {
+    if (trigger.world.destinations.find(destination) != trigger.world.destinations.end()) {
+      const radix::Quaternion &destination_orientation = trigger.world.destinations.at(destination).orientation;
+      const radix::Quaternion &old_player_orientation = player.getHeadOrientation();
+
+      /*
+       * destination_orientation : describes the rotation needed to tranform a vector in the
+       * world coordinates to a vector following the orientation of the destination portal.
+       *
+       * old_player_orientation : describes the rotation to transform a reference vector into a vector
+       * following the player orientation
+       *
+       * The product of two quaternions can be seen as the composition of two rotations.
+       * (all quaternions can be interpreted as a rotation)
+       */
+      radix::Quaternion new_player_orientation = destination_orientation * old_player_orientation;
```

Comment:
new_player_orientation is inconsistant with our naming convention. We use camel case for varible names.